### ISSUE TITLE
ceph: we need to call sgdisk --clear during OSD mkfs 

### ISSUE THREAD

**Issue by bassam** [rook/rook#327]
This matches what `ceph-disk` does. Without it we believe that bluestore gets into a weird state where the superblock is not read correctly:

```
Core was generated by `/usr/bin/rookd daemon --type=osd -- --foreground --id=15 --cluster=rookcluster'.
Program terminated with signal SIGABRT, Aborted.
#0  raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
58  ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
[Current thread is 1 (Thread 0xc7ee240 (LWP 2074))]
warning: Missing auto-load script at offset 0 in section .debug_gdb_scripts
of file /media/root/tmp/rookd-extract/rootfs/usr/bin/rookd.
Use `info auto-load python-scripts [REGEXP]' to list them.
(gdb) bt
#0  raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
#1  0x000000000185bccc in reraise_fatal (signum=6) at /home/rook/go/src/github.com/rook/rook/ceph/src/global/signal_handler.cc:72
#2  handle_fatal_signal (signum=6) at /home/rook/go/src/github.com/rook/rook/ceph/src/global/signal_handler.cc:134
#3  <signal handler called>
#4  raise (sig=<optimized out>) at ../sysdeps/unix/sysv/linux/raise.c:58
#5  0x00000000022cb37a in abort ()
#6  0x000000000226dafd in __gnu_cxx::__verbose_terminate_handler() ()
#7  0x00000000021df496 in __cxxabiv1::__terminate(void (*)()) ()
#8  0x00000000021df4e1 in std::terminate() ()
#9  0x00000000021dfdc9 in __cxa_throw ()
#10 0x0000000001da6483 in bluefs_super_t::decode (this=this@entry=0xd40e0e8, p=...) at /home/rook/go/src/github.com/rook/rook/ceph/src/os/bluestore/bluefs_types.cc:73
#11 0x0000000001a699c3 in decode (p=..., c=...) at /home/rook/go/src/github.com/rook/rook/ceph/src/os/bluestore/bluefs_types.h:77
#12 BlueFS::_open_super (this=this@entry=0xd40e000) at /home/rook/go/src/github.com/rook/rook/ceph/src/os/bluestore/BlueFS.cc:458
#13 0x0000000001a80081 in BlueFS::mount (this=0xd40e000) at /home/rook/go/src/github.com/rook/rook/ceph/src/os/bluestore/BlueFS.cc:352
#14 0x0000000001a175ea in BlueStore::_open_db (this=this@entry=0xd192000, create=create@entry=false) at /home/rook/go/src/github.com/rook/rook/ceph/src/os/bluestore/BlueStore.cc:3324
#15 0x0000000001a48dbc in BlueStore::mount (this=0xd192000) at /home/rook/go/src/github.com/rook/rook/ceph/src/os/bluestore/BlueStore.cc:3983
#16 0x00000000016d2cc7 in OSD::init (this=0xd2ee000) at /home/rook/go/src/github.com/rook/rook/ceph/src/osd/OSD.cc:2042
#17 0x00000000014b9547 in cephd_osd (argc=<optimized out>, argv=<optimized out>) at /home/rook/go/src/github.com/rook/rook/ceph/src/ceph_osd.cc:611
#18 0x0000000000dfa9f5 in cephd_run_osd (argc=<optimized out>, argv=<optimized out>) at /home/rook/go/src/github.com/rook/rook/ceph/src/libcephd/libcephd.cc:237
#19 0x0000000000c1191a in _cgo_49d41b80151a_Cfunc_cephd_run_osd (v=0xc4201c1b28) at /home/rook/go/src/github.com/rook/rook/pkg/cephmgr/cephd/cephd.go:204
#20 0x00000000004f36f0 in runtime.asmcgocall () at /usr/local/go/src/runtime/asm_amd64.s:590
#21 0x000000000049846d in runtime.cgocall (fn=0x7ffe9cab72a0, arg=0x325d580 <runtime.g0>, ~r2=-1666485616) at /usr/local/go/src/runtime/cgocall.go:115
#22 0x000000000325d500 in runtime.work ()
#23 0x00007ffe9cab72a0 in ?? ()
#24 0x000000000325d580 in ?? ()
#25 0x00007ffe9cab7290 in ?? ()
#26 0x00000000004c62e4 in runtime.mstart () at /usr/local/go/src/runtime/proc.go:1096
#27 0x00000000004f1845 in runtime.rt0_go () at /usr/local/go/src/runtime/asm_amd64.s:156
#28 0x000000000000000b in ?? ()
#29 0x00007ffe9cab73f8 in ?? ()
#30 0x000000000000000b in ?? ()
#31 0x00007ffe9cab73f8 in ?? ()
#32 0x0000000000400598 in __rela_iplt_start ()
#33 0x00000000022ba2f6 in generic_start_main ()
#34 0x00000000022ba57e in __libc_start_main ()
#35 0x000000000049307a in _start ()
```

**Comment by bassam** [comment]
Related http://tracker.ceph.com/issues/18389

**Comment by travisn** [comment]
Fixed by #330
